components/ContinentScreen/ContinentScreen.jsx

```diff
@@ -0,0 +1,123 @@
+import { ActivityIndicator, ScrollView, Text, View, Pressable } from "react-native";
+import { useLocalSearchParams, useRouter } from "expo-router";
+
+import { theme } from "../../utils/theme";
+import BackgroundView from "../BackgroundView";
+import useFetchEarth from "../../utils/hooks/useFetchEarth";
+
+
+function ContinentScreen() {
+  const params = useLocalSearchParams();
```

Comment:
move these hooks and api calls out into a separate datalayer. This allows you to keep the display layer pure and easy to test by changing the props alone.